### PR TITLE
Clarify python versions needed

### PR DESCRIPTION
**Pull request by ReubenHill**

### CODE REVIEW

File: docs/source/download.rst [hunk 1]
```diff
@@ -98,7 +98,7 @@ packages can be installed into an existing Firedrake installation using
 System requirements
 -------------------
 
-Firedrake requires Python 3.6 or later. The installation script is
+Firedrake requires Python 3.6.x to 3.8.x. The installation script is
```
Comment (wence-) [inline docs/source/download.rst:101]: Add a note about lack of binary wheel for Pyhton 3.9 packages?

**Comment by wence-** [comment]
I merged the (hopeful) test fix. Please rebase on top of master.

File: docs/source/download.rst (hunk 1 above)
Comment (ReubenHill) [reply docs/source/download.rst]: Changed in rebased commit 9381afe

**Review by wence- — APPROVED** [review]
Thanks, will merge.

**Pull request merged**